ee/api/dashboard_collaborator.py

```diff
@@ -63,9 +63,9 @@ def create(self, validated_data):
             raise serializers.ValidationError("User does not exist.")
         if cast(Team, dashboard.team).get_effective_membership_level(validated_data["user"].id) is None:
             raise exceptions.ValidationError("Cannot add collaborators that have no access to the project.")
-        if dashboard.does_user_have_inherent_restriction_rights(validated_data["user"].id):
+        if dashboard.can_user_restrict(validated_data["user"].id):
             raise exceptions.ValidationError(
-                "A user with inherent dashboard restriction rights (the dashboard owner or a project admins) cannot be added as a collaborator."
+                "A user with dashboard restriction rights (the dashboard owner or a project admins) cannot be added as a collaborator."
```

Comment:
This message feels hard to understand. Are we saying: something like: "You don't need to add this person as a collaborator they already have access"?